lib/resolveGitRepo.js

```diff
@@ -14,7 +14,9 @@ const debugLog = debug('lint-staged:resolveGitRepo')
  * submodules and worktrees
  */
 const resolveGitConfigDir = async (gitDir) => {
-  const defaultDir = normalize(path.join(gitDir, '.git'))
+  // Get the real path in case it's a symlink
+  const realpath = await fs.realpath(path.join(gitDir, '.git'))
+  const defaultDir = normalize(realpath)
```

Comment:
Sorry, maybe just:

```suggestion
  // Get the real path in case it's a symlink
  const defaultDir = normalize(await fs.realpath(path.join(gitDir, '.git')))
```